File: Baekjoon/Class2/35.18110 solved.ac.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    cin >> n;
    if (n == 0) {
        cout << '0';
        return 0;
    }

    vector<int> opinions(n);
    for (int i = 0; i < n; i++)
        cin >> opinions[i];
    sort(opinions.begin(), opinions.end()); // 정렬

    int total = 0;
    int exclusion = int(round(n * 0.15)); // 절사 범위
    for (int i = exclusion; i < n - exclusion; i++) // 절사
        total += opinions[i];
    cout << round(double(total) / (n - 2 * exclusion)); // 평균
    return 0;
}

File: Baekjoon/Class2/35.18110 solved.ac.java
import java.io.*;
import java.util.Arrays;

class Main {
    public static void main(String[] args) throws IOException {
        int n;
        int[] opinions;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());
        if(n == 0)
            System.out.print(0);
        else {
            opinions = new int[n];
            for(int i = 0; i < n; i++)
                opinions[i] = Integer.parseInt(br.readLine());
            Arrays.sort(opinions); // 정렬

            int total = 0;
            int exclusion = (int)Math.round(n * 0.15); // 절사 범위
            for(int i = exclusion; i < n - exclusion; i++) // 절사
                total += opinions[i];
            System.out.print(Math.round((double)total / (n - 2 * exclusion))); // 평균
        }
        br.close();
    }
}